server/modules/category/api/categoryApi.js

```diff
@@ -96,23 +96,61 @@ export const init = app => {
       } catch (error) {
         next(error)
       }
-    },
+    }
   )
 
   // ==== READ
 
-  app.get('/survey/:surveyId/categories', AuthMiddleware.requireSurveyViewPermission, async (req, res, next) => {
+  app.get('/survey/:surveyId/categoriesAll', AuthMiddleware.requireSurveyViewPermission, async (req, res, next) => {
     try {
       const { surveyId, draft, validate } = Request.getParams(req)
 
-      const categories = await CategoryService.fetchCategoriesAndLevelsBySurveyId(surveyId, draft, validate)
+      const categories = await CategoryService.fetchCategoriesAndLevelsBySurveyId({
+        surveyId,
+        draft,
+        includeValidation: validate,
+      })
 
       res.json({ categories })
     } catch (error) {
       next(error)
     }
   })
 
+  app.get('/survey/:surveyId/categories', AuthMiddleware.requireSurveyViewPermission, async (req, res, next) => {
+    try {
+      const { surveyId, draft, validate, offset = 0, limit = null } = Request.getParams(req)
+
+      const list = await CategoryService.fetchCategoriesBySurveyId({
+        surveyId,
+        draft,
+        includeValidation: validate,
```

Comment:
maybe the param of CategoryService.fetchCategoriesBySurveyId can also be called validate

Reply:
I prefer to keep it as it is, it does not actually validate but only fetch the categories with validation. Maybe the parameter name can be switched between the service layer and the manager.